### PR TITLE
Optimize exit game

### LINKED ISSUE #91: submit with block number
## Requirements

- We wanna know which deposit block or normal block a block is by block number in exit game
- because inclusion check is different

### Other works

In loom network plasma cash, 

- submit merkle root with block number
- normal blocks are by 1000 num
- deposit blocks are  by 1 num
- We can check by 'blkNum % 1000 != 0'

Refrrences: https://github.com/loomnetwork/plasma-cash/blob/master/server/contracts/Core/RootChain.sol#L241

### LINKED ISSUE #119: Oprimize exit-game
## Description

We can reduce the cost of `startExit` by dividing exit-game.

- [x] Update root chain
- [ ] Update wallet

https://scrapbox.io/cryptoeconimicslab/Optimize_Exit_Process

### PR DESCRIPTION
**Pull request by syuhei176**
## Description

Optimize exit-game.
Make RootChain contract more simple.

- [x] Reduce gas cost of start Exit. (current cost of simple state exit is 454,112)
- [x] Check deposit transaction
- [x] Fix bug of exiting the transaction which is the child of deposit.
- [ ] double spent challenge

Closes #91 
Closes #119

### CODE REVIEW

File: packages/chamber-rootchain/contracts/RootChain.sol [hunk 1]
```diff
@@ -369,33 +307,44 @@ contract RootChain {
...
+      bytes _exitTxBytes,
+      bytes _txInfoBytes
     )
       public
     {
       var childBlock = childChains[_chain].blocks[_blkNum];
-      var txList = parseExitTxList(childChains[_chain], _blkNum, _txListBytes);
-      var output = txList[txList.length - 1].tx.outputs[txList[txList.length - 1].index];
+      var exitTx = TxVerification.getTx(_exitTxBytes);
+      // if there are 2 inputs this exit don't need prev tx because it has confsig.
```
[the remark this answers is not in the document]
Comment (shogochiai) [reply packages/chamber-rootchain/contracts/RootChain.sol]: I can't see the separated another `startExit`? Or I fundamentally misunderstood?

[the remark this answers is not in the document]
Comment (syuhei176) [reply packages/chamber-rootchain/contracts/RootChain.sol]: Please check this.
https://github.com/cryptoeconomicslab/plasma-chamber/issues/119#issuecomment-445709932

**Comment by syuhei176** [comment]
Fix some to support this case.

**Comment by syuhei176** [comment]
The picture of the latest spec.

**Pull request merged**